python/paddle/fluid/dygraph/dygraph_to_static/break_continue_transformer.py

```diff
@@ -282,8 +265,100 @@ def _add_stmt_into_list_before_node(self, stmt_list, node, stmt_node):
         stmt_list.insert(i, stmt_node)
         return True
 
-    def _find_ancestor_loop_index(self, node):
-        for i in range(len(self.ancestor_nodes) - 1, -1, -1):
-            if isinstance(self.ancestor_nodes[i], (gast.For, gast.While)):
-                return i
-        return -1
+
+def _find_ancestor_loop_index(node, ancestor_nodes):
+    for i in range(len(ancestor_nodes) - 1, -1, -1):
+        if isinstance(ancestor_nodes[i], (gast.For, gast.While)):
+            return i
+    return -1
+
+
+class BreakTransformOptimizer(BaseNodeVisitor):
+    """
+    In some pattern, the transformed code could be optimized by joining the 
```

Comment:
Had better to say what pattern it is, then code readers can know what this class optimizes immediately. From your code, I think it is "If there is a conditional 'break' statement right after loop".

Reply:
Thanks, fixed.